src/Bicep.Core/TypeSystem/TypeAssignmentVisitor.cs

```diff
@@ -1056,6 +1050,18 @@ private void ValidateDecorators(IEnumerable<DecoratorSyntax> decoratorSyntaxes,
 
                     if (decorator is not null)
                     {
+                        for (int i = 0; i < decoratorSyntax.Arguments.Length; i++)
+                        {
+                            var parameterFlags = decorator.Overload.FixedParameters.Length < i
```

Comment:
Should this be `>`?

Reply:
Yes, but it was actually redundant to the same check performed in the [`GetFunctionSymbolType` method](https://github.com/Azure/bicep/pull/17804/files#diff-146833b5e46faf0de8cb0cbdd86fa5d041c342ca4480c187f5c74af6044afcd0R2546). The incorrect comparator was keeping this passage from generator duplicate diagnostics.

Removed in the latest commit.